### PR TITLE
[ShadCN] feat: create table component set with ShadCN

### PR DESCRIPTION
**Pull request by TylerAPfledderer**
## Description

Creates the Table components using ShadCN.

Creates a separate story file until deemed ready to merge and be a part of the Chromatic snapshots.

Also introduces [tailwind-variants](https://www.tailwind-variants.org/) which allows you to create [slot variants](https://www.tailwind-variants.org/docs/slots): variant sets to pass to a multi-component (i.e. `Table`, `Tag`, etc. that has many parts)

`tailwind-variants` to be used in place of `class-variance-authority`

### CODE REVIEW

File: src/components/ui/__stories__/Table/Table.tsx
Comment (pettinarip) [inline src/components/ui/__stories__/Table/Table.tsx:1]: This file should be outside the `__stories__` folder

File: src/components/ui/__stories__/Table/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+/**
+ * TODO: Currently, there are cell spacing issues with some table content.
+ * Prefer `table-fixed` utility class in the future when content has been addressed
+ * to provide equal cell widths.
+ */
+
+const tableVariants = tv({
+  slots: {
+    table: "w-full",
+    th: "text-start border-b-[1px] border-body text-body normal-case align-bottom p-4",
```
Comment (pettinarip) [inline src/components/ui/__stories__/Table/Table.tsx:16]:
```suggestion
    th: "text-start border-b border-body text-body normal-case align-bottom p-4",
```

File: src/components/ui/__stories__/Table/mockMdxData.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
+import * as React from "react"
+
+import InlineLink from "../../../../../tailwind/ui/Link"
```
Comment (pettinarip) [inline src/components/ui/__stories__/Table/mockMdxData.tsx:3]: should be `import InlineLink from "@/components/Link"`

**Comment by TylerAPfledderer** [comment]
@pettinarip did a full replacement of the table component. So this new version is available to be checked for prod.

**Review by pettinarip — APPROVED** [review]
LGTM @TylerAPfledderer great job!

**Review by nloureiro — APPROVED** [review]

**Pull request merged**